### ISSUE TITLE
sub-segment travel times 

### ISSUE THREAD

**Issue by kpwebb** [opentraffic/traffic-data-exchange-format#5]
Currently we're segmenting the street network by intersections which works well in terms of integration with graph representations in OTP (and I assume OSRM) because every intersection should contain an OSM node that we can use as a reference. 

However, if we want to create sub-segments of the network for long sections between intersections, how will this align with the representations in a journey planner? Should we store the sub-segment speeds with in the intersection segment and use linear referencing to define the sub-segment? We can count on segments having any intermediate nodes (could just be a long straight line with a pair of nodes at each end) and even if they do the nodes may not align with where we want to segment.

I think we can add this incrementally to the existing spec but curious if there are examples to draw from our thoughts on how this could be useful in a routing context. I'm guessing the primary application is visualization or routes that terminate mid-segment.

**Comment by laurentg** [comment]
A linear referencing attached to the "parent" segment is indeed the easier to use from a route planner point of view. The linear reference could be either in some fixed unit (float / integer meters) or scaled (normalized) to the parent length, from 0 to 1.

Using intermediate node ID's do have indeed lots of drawback. Unless we have a stable and reproducible algorithm to split segments, it would be difficult for both sides to determine which node to use in the index process. Obviously, we could keep _all_ intermediate node ID's in the index, but that would be somehow expensive in term of memory.

**Comment by abyrd** [comment]
Agreed with @laurentg that linear referencing along the street segment between two intersections is probably the way to go. I'd tend toward normalized to the parent segment length (i.e. values between 0 and 1) but that's an implementation detail.
